src/utils/counter.ts

```diff
@@ -0,0 +1,15 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END=
+
+const counter = () => {
+    let _counter = 0;
```

Comment:
I think `counter` should be reset to zero everytime a new search is performed. This will help debugging (i.e. checking DOM or other IDs), but also avoid super long numbers when running many searches in the same Thorium executable session.

Reply:
I don't reset counter to 0 specially to avoid a collision. For example If the remove highlight procedure fail between 2 search word.
It can be also useful to have a unique id number in the window for the highlight annotation or whatever unique number.

Reply:
I can also add a reset test on Number.MAX_SAFE_INTEGER condition

Reply:
https://github.com/edrlab/thorium-reader/pull/1057/commits/e69a0c78448197d888cb6580631f67c947ca3058